src/lint/rules/mod.rs

```diff
@@ -26,34 +35,48 @@ pub fn  instantiate_rules(cfg: &LintCfg) -> CurrentRules {
         nsp_inparen: NspInparenRule { enabled: cfg.nsp_inparen.is_some() },
         nsp_unary: NspUnaryRule { enabled: cfg.nsp_unary.is_some() },
         nsp_trailing: NspTrailingRule { enabled: cfg.nsp_trailing.is_some() },
-        long_lines: LongLineOptions::into_rule(&cfg.long_lines),
+        long_lines: LongLinesRule::from_options(&cfg.long_lines),
+        indent_no_tabs: IndentNoTabRule { enabled: cfg.indent_no_tabs.is_some() },
+        indent_code_block: IndentCodeBlockRule::from_options(&cfg.indent_code_block),
+        indent_closing_brace: IndentClosingBraceRule::from_options(&cfg.indent_closing_brace),
+        indent_paren_expr: IndentParenExprRule { enabled: cfg.indent_paren_expr.is_some() },
+        indent_switch_case: IndentSwitchCaseRule::from_options(&cfg.indent_switch_case),
+        indent_empty_loop: IndentEmptyLoopRule::from_options(&cfg.indent_empty_loop)
     }
 }
 
 // struct/trait generic_rule
 pub trait Rule {
     fn name() -> &'static str;
     fn description() -> &'static str;
+    fn get_rule_type() -> RuleType;
+    fn push_err(&self, acc: &mut Vec<DMLStyleError>, range: ZeroRange) {
```

Comment:
Function shouldn't both create the error and add it to an accumulator (even if the accumulator is the common use-case), cleaner to just do a function "create_err" or similar and leave the accumulating to the caller.

Reply:
Changed to create_err() to have a single responsibility in this function. Push is done on each caller.